catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/forms/SearchFormsApplicationTest.java

```diff
@@ -0,0 +1,438 @@
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.mockito.ArgumentCaptor;
+import org.opengis.filter.Filter;
+import spark.Spark;
+
+/**
+ * Spin up the {@link SearchFormsApplication} spark application and verify the REST contract against
+ * a mocked {@link CatalogFramework}.
+ *
+ * <p>This test suite uses HTTP to verify one of the data contracts for catalog-ui-search. By
+ * testing the network directly, the tests remain stable regardless of the JSON solution used.
+ *
+ * <p>This particular test suite verifies both directions of JSON and XML transforms using
+ * parameterized REST messages that are interpolated with actual test values at runtime. The
+ * objective of this test suite is to ensure serialization can handle any special character.
+ */
+@RunWith(Parameterized.class)
+public class SearchFormsApplicationTest {
+  /**
+   * Set of symbols upon which to parameterize the tests; defined as they appear in a JSON document.
+   */
+  @Parameterized.Parameters(name = "Verify search form REST I/O for symbol: {0}")
+  public static Iterable<Object[]> data() {
+    return Arrays.asList(
+        new Object[][] {
+          {"'"}, {"\\\""}, {"\\\\"}, {">"}, {"<"}, {"&"}, {"{"}, {"}"}, {"["}, {"]"}, {":"}, {";"},
+          {","}, {"."}, {"?"}, {"/"}, {"|"}, {"-"}, {"_"}, {"+"}, {"="}, {"*"}, {"^"}, {"%"}, {"$"},
+          {"#"}, {"@"}, {"!"}, {"~"}, {"`"}, {"("}, {")"}
+        });
+  }
+
+  /**
+   * Not all JSON representations are communitively validatable against XML. This mapping is what
+   * all the tests use to ensure expected XML counterparts are built correctly.
+   */
+  private static final Map<String, String> JSON_TO_XML_SYMBOL_MAPPING =
+      ImmutableMap.of(
+          "\\\"", "\"",
+          "\\\\", "\\",
+          ">", "&gt;",
+          "<", "&lt;",
+          "&", "&amp;");
+
+  private static final URL FILTER_RESOURCES_DIR =
+      SearchFormsLoaderTest.class.getResource("/forms/app");
+
+  private static final Header CONTENT_IS_JSON = new Header("Content-Type", "application/json");
+
+  private static final String TEMPLATE_FORM_METACARD_JSON_SIMPLE =
+      getContentsOfFile("form-simple.json");
+
+  private static final String TEMPLATE_FORM_METACARD_JSON_SIMPLE_RESPONSE =
+      removePrettyPrintingOnJson(getContentsOfFile("form-simple-response.json"));
+
+  private static final String TEMPLATE_FORM_FILTER_XML_SIMPLE =
+      removePrettyPrintingOnXml(getContentsOfFile("form-filter-simple.xml"));
+
+  private static final String CANNED_TITLE = "MY_TITLE";
+
+  private static final String CANNED_DESCRIPTION = "MY_DESCRIPTION";
+
+  private static final String CANNED_ID = "abcdefg";
+
+  private static final String CANNED_ISO_DATE = "2018-12-10T13:09:40Z";
+
+  private static final FilterBuilder MOCK_BUILDER = new GeotoolsFilterBuilder();
```

Comment:
✏️ It was actually easier to just use the real dependency here to simplify interactions with `EndpointUtil`. I can remove the `MOCK` prefix since that doesn't really make sense anymore, but I'm open to other ideas.